### PR TITLE
AEA-2527 Population of Prescriber Codes

### PR DESCRIPTION
**Pull request by TomSmithCoding**
AC 1.1
WHEN a GMP code is the only professional identifier that is provided for a prescriber,
THEN the FHIR Facade API needs to reject this
AND request that a more suitable identifier, e.g. a GMC Reference Number, is provided for the creation of a prescription

### CODE REVIEW

File: coordinator/tests/services/validation/bundle-validator.spec.ts [hunk 1]
```diff
@@ -446,6 +449,16 @@ describe("MedicationRequest consistency checks", () => {
...
 
+  test("Should throw error when GMP-number is only number in Practitioner.identifier", () => {
+    const testReference: fhir.Identifier = createIdentifier("https://fhir.hl7.org.uk/Id/gmp-number", "G1234567")
+    if (isReference(practitionerRoles[0].practitioner)) {
+      const practitioner = resolveReference(bundle, practitionerRoles[0].practitioner)
+      practitioner.identifier = [testReference]
+    }
+
+    const validationErrors = validator.verifyPrescriptionBundle(bundle)
+    expect(validationErrors).toHaveLength(1)
```
Comment (AndyMather24) [inline coordinator/tests/services/validation/bundle-validator.spec.ts:460]: Hi @TomSmithCoding,  We could check the error details in the expect statement to ensure we are throwing the correct error for this scenario.

Comment (TomSmithCoding) [reply coordinator/tests/services/validation/bundle-validator.spec.ts:460]: Sounds good! Good to be specific, will get this added.

**Review by AndyMather24 — DISMISSED** [review]

File: coordinator/tests/services/validation/bundle-validator.spec.ts [hunk 2]
```diff
@@ -15,6 +16,8 @@ import {
   PRESCRIBING_APP_SCOPE,
   PRESCRIBING_USER_SCOPE
 } from "../../../src/services/validation/scope-validator"
+import {isReference} from "../../../src/utils/type-guards"
+import {createIdentifier} from "../../../../models/fhir"
```
Comment (StefanoFrazzetto) [inline coordinator/tests/services/validation/bundle-validator.spec.ts:20]: Remove import and use fhir.createIdentifier below

File: coordinator/tests/services/validation/bundle-validator.spec.ts [hunk 3]
```diff
@@ -446,6 +449,18 @@ describe("MedicationRequest consistency checks", () => {
     expect(validationErrors).toHaveLength(1)
   })
 
+  test("Should throw error when GMP-number is only number in Practitioner.identifier", () => {
+    const testReference: fhir.Identifier = createIdentifier("https://fhir.hl7.org.uk/Id/gmp-number", "G1234567")
```
Comment (StefanoFrazzetto) [inline coordinator/tests/services/validation/bundle-validator.spec.ts:453]: You can replace this with fhir.createIdentifier and remove the additional import

**Review by StefanoFrazzetto — APPROVED** [review]
Great job! :)

**Pull request merged**